pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -43,7 +68,7 @@ static boolean castToBoolean(final Object in) throws PSQLException {
       return fromString((String) in);
     }
     if (in instanceof Character) {
-      return fromCharacter((Character) in);
+      return fromString(in.toString());
```

Comment:
This peaked my curiosity, so I threw together a quick jmh benchmark. It turns out you were right to be concerned, but not because of the `toString` call. The identity comparisons of chars (even with auto-unboxing) is simply significantly faster than string comparisons.
However, a switch statement is even slightly faster. So I will be providing that.